homeassistant/components/github/__init__.py

```diff
@@ -38,7 +38,9 @@ async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
         )
 
         await coordinator.async_config_entry_first_refresh()
-        await coordinator.subscribe()
+
+        if not entry.pref_disable_polling:
```

Comment:
What if the user enables it again? How would it subscribe?

Reply:
If a user does that, the config entry will reload and it will subscribe

Reply:
https://github.com/home-assistant/core/commit/95a1ecb0dca7255b4d6f04f45c592957d1e317b0

Reply:
Sure that tests works, as it calls:

```
await hass.config_entries.async_reload(mock_config_entry.entry_id)
```

in the test.

But when a user changes a system config option, the integration isn't reloaded? Or is it?

Reply:
edit: Me stupid, 

The integration does:

`entry.async_on_unload(entry.add_update_listener(async_reload_entry))`

Nevermind me 🙈